mockld/streaming_service.go

```diff
@@ -123,29 +129,66 @@ func (s *StreamingService) makePutEvent() eventsource.Event {
 	}
 }
 
+// Sends on SSE event to all clients that are currently connected to the stream-- or, if no client
+// has connected yet, queues the event so that it will be sent (after the initial data) to the
+// first client that connects. (The latter is necessary to avoid race conditions, since even after
+// a connection is received on the stream endpoint, it is hard for the test logic to know when the
+// HTTP handler has actually created a stream subscription for that connection. If we called
+// the eventsource Publish method before a subscription existed, the event would be lost.)
```

Comment:
The logic here depends on implementation details of `eventsource.Server`, since that's what we're using to actually publish SSE streams. What actually happens when a stream request comes in from the SDK is:

1. The test harness's router figures out what MockEndpoint it corresponds to.
2. It records that we have received a connection.
3. It calls the `eventsource.Server`'s HTTP handler.
4. That handler calls our `ReplayAll` method to provide any _initial_ events for the stream.
5. Then the handler it leaves the connection open, and enters a loop where it sends any events that we choose to publish after that point.

Without the extra synchronization that I added here, it was easy to end up publishing the event after 2 but before 5.